project_timesheet_currency/tests/test_multicurrency_timesheet.py

```diff
@@ -0,0 +1,150 @@
+
+        # I create analytic lines for every day of the week
+        # passing project and task to simulate the creation of analytic lines
+        # through the task form
+        al_model = self.env['account.analytic.line']
+        date_05_08 = time.strftime('%Y-05-08')
+        al_model.sudo(self.user_fr.id).with_context(
+            {'date': date_05_08}).create(
+            {
+                'name': 'Test USD',
+                'project_id': self.project_USD.id,
+                'task_id': self.task_USD.id,
+                'account_id': self.project_USD.analytic_account_id.id,
+                'unit_amount': 8,
+                'date': date_05_08,
+            })
+        date_05_09 = time.strftime('%Y-05-09')
+        al_model.sudo(self.user_fr.id).with_context(
+            {'date': date_05_09}).create(
+            {
+                'name': 'Test USD',
+                'project_id': self.project_USD.id,
+                'task_id': self.task_USD.id,
+                'account_id': self.project_USD.analytic_account_id.id,
+                'unit_amount': 8,
+                'date': date_05_09,
+            })
+        date_05_10 = time.strftime('%Y-05-10')
+        al_model.sudo(self.user_fr.id).with_context(
+            {'date': date_05_10}).create(
+            {
+                'name': 'Test USD',
+                'project_id': self.project_USD.id,
+                'task_id': self.task_USD.id,
+                'account_id': self.project_USD.analytic_account_id.id,
+                'unit_amount': 8,
+                'date': date_05_10,
+            })
+        date_05_11 = time.strftime('%Y-05-11')
+        al_model.sudo(self.user_fr.id).with_context(
+            {'date': date_05_11}).create(
+            {
+                'name': 'Test USD',
+                'project_id': self.project_USD.id,
+                'task_id': self.task_USD.id,
+                'account_id': self.project_USD.analytic_account_id.id,
+                'unit_amount': 8,
+                'date': date_05_11,
+            })
+        date_05_12 = time.strftime('%Y-05-12')
+        al_model.sudo(self.user_fr.id).with_context(
+            {'date': date_05_12}).create(
+            {
+                'name': 'Test USD',
+                'project_id': self.project_USD.id,
+                'task_id': self.task_USD.id,
+                'account_id': self.project_USD.analytic_account_id.id,
+                'unit_amount': 8,
+                'date': date_05_12,
+            })
+
+        a_lines = self.project_USD.analytic_account_id.line_ids
+        amounts = [l.amount for l in a_lines]
+        amounts_currency = [l.amount_currency for l in a_lines]
+        total_amount = sum(amounts)
+        total_amount_currency = sum(amounts_currency)
+
+        # I check that the currency conversion worked as expected and both
+        # amount and amount_currency fields values are correct
+        self.assertEqual(total_amount, -4734.5199999999995)
```

Comment:
better use `self.assertAlmostEqual(...)` see here https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertAlmostEqual

same for line 150